File: frontend/src/types/project.ts
import type { Property } from './deal';
import type { ParsedDocument } from './document';

export interface ChatSessionBrief {
  id: string;
  title: string | null;
  analysis_type: string | null;
  message_count: number;
  created_at: string;
  updated_at: string;
}

export interface Project {
  id: string;
  user_id: string;
  property_id: string | null;
  name: string;
  description: string | null;
  instructions: string | null;
  is_archived: boolean;
  document_count: number;
  session_count: number;
  created_at: string;
  updated_at: string;
}

export interface ProjectDetail extends Project {
  property: Property | null;
  documents: ParsedDocument[];
  sessions: ChatSessionBrief[];
}

export interface ProjectListResponse {
  items: Project[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProjectCreate {
  name: string;
  property_id?: string;
  description?: string;
  instructions?: string;
}

export interface ProjectUpdate {
  name?: string;
  property_id?: string;
  description?: string;
  instructions?: string;
}

File: backend/app/models/project.py
"""
Pydantic models for the Projects API.
"""
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.deal import PropertyResponse
from app.models.document import ParsedDocumentResponse


class ProjectCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    property_id: str | None = None
    description: str | None = None
    instructions: str | None = None
    property_address: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    property_id: str | None = None
    description: str | None = None
    instructions: str | None = None
    property_address: str | None = None


class ChatSessionBrief(BaseModel):
    id: UUID
    title: str | None = None
    analysis_type: str | None = None
    message_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectResponse(BaseModel):
    id: UUID
    user_id: UUID
    property_id: UUID | None = None
    name: str
    description: str | None = None
    instructions: str | None = None
    property_address: str | None = None
    is_archived: bool = False
    document_count: int = 0
    session_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectDetailResponse(ProjectResponse):
    property: PropertyResponse | None = None
    documents: list[ParsedDocumentResponse] = []
    sessions: list[ChatSessionBrief] = []


class ProjectListResponse(BaseModel):
    items: list[ProjectResponse]
    total: int
    page: int
    page_size: int
